pkg/repo/repo_ent_impl.go

```diff
@@ -136,37 +138,84 @@ func (e *entImpl) Close() {
 	e.entClient.Close()
 }
 
-func (e *entImpl) HandleError(ctx context.Context, err error) error {
+func (e *entImpl) HandleError(err error) error {
+	if err == nil {
+		return nil
+	}
+
 	if ent.IsNotFound(err) {
-		return NewNotFoundError(err)
+		return NotFoundError{Err: err}
 	}
 	if ent.IsValidationError(err) {
-		return NewValidationError(err, "Validation error")
-	}
-	if _, ok := err.(govalidator.Errors); ok {
-		return NewValidationError(err, "Validation error")
-	}
-	if _, ok := err.(validator.ValidationErrors); ok {
-		return NewValidationError(err, "Validation error")
+		return ValidationError{Err: err, Message: "Validation error"}
 	}
 	if ent.IsConstraintError(err) {
 		if strings.Contains(err.Error(), "UNIQUE constraint failed: facts.hashed_value, facts.scope_facts, facts.fact_type_facts") {
```

Comment:
@jperichon what's the error message from postgres? I mean if we get `ent.IsConstraintError(err)`, maybe it's safe to assume it's a `ValidationError` without actually checking the string contains - at least for fact creation.

Reply:
From postgres:

```
"ent: constraint failed: insert node to table \"facts\": pq: duplicate key value violates unique constraint \"fact_hashed_value_scope_facts_fact_type_facts\"""
```

For sqlite:
```
"ent: constraint failed: insert node to table \"facts\": UNIQUE constraint failed: facts.hashed_value, facts.scope_facts, facts.fact_type_facts"
```

I made an attempt for fix at: https://github.com/open-privacy/opv/pull/33